jsonschema2pojo-integration-tests/src/test/java/org/jsonschema2pojo/integration/CompilerWarningIT.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * <p>Notes: The eclipse compiler used in these tests has an open issue with the SuppressWarnings annotation.  As a result, some warnings must be
+ * accepted here that would not be present in practice.
+ * <a href="https://bugs.eclipse.org/bugs/show_bug.cgi?id=469725">Bug 469725 - ECJ compiler: @SuppressWarnings annotation is ignored when ecj is invoked via java compiler tool API</a>
+ * </p>
+ *
+ * @author Christian Trimble
+ *
+ */
+@RunWith(Parameterized.class)
+public class CompilerWarningIT {
+  @Rule public Jsonschema2PojoRule schemaRule = new Jsonschema2PojoRule();
+  
+  @Parameters(name="{0}")
+  public static Collection<Object[]> parameters() {
+    return Arrays.asList(new Object[][] {
+      {
+        "includeAccessorsWithSystemJavaCompiler",
+        Compiler.systemJavaCompiler(),
+        config("includeDynamicAccessors", true),
+        "/schema/dynamic/parentType.json",
+        Matchers.empty()
+      },
+      {
+        "includeAccessorsWithEclipseCompiler",
+        Compiler.eclipseCompiler(),
+        config("includeDynamicAccessors", true),
+        "/schema/dynamic/parentType.json",
+        onlyCastExceptions()
+      }
+    });
+  }
+
+  private CompilerSupplier supplier;
+  private Map<String, Object> config;
+  private String schema;
+  private Matcher<List<Diagnostic<? extends JavaFileObject>>> matcher;
+
+  public CompilerWarningIT(String label, CompilerSupplier supplier, Map<String, Object> config, String schema, Matcher<List<Diagnostic<? extends JavaFileObject>>> matcher) {
+    this.supplier = supplier;
+    this.config = config;
+    this.schema = schema;
+    this.matcher = matcher;
+  }
+
+  @Test
+  public void checkWarnings() {
+    schemaRule.generate(schema, "com.example", config);
+    schemaRule.compile(supplier, new NullWriter(), new ArrayList<File>(), config);
+    
+    List<Diagnostic<? extends JavaFileObject>> warnings = warnings(schemaRule.getDiagnostics());
+    
+    assertThat(warnings, matcher);
+  }
+
+  public static List<Diagnostic<? extends JavaFileObject>> warnings(List<Diagnostic<? extends JavaFileObject>> all) {
+    List<Diagnostic<? extends JavaFileObject>> warnings = new ArrayList<>();
+    for( Diagnostic<? extends JavaFileObject> entry : all ) {
+      if( entry.getKind() == Kind.WARNING ) {
+        warnings.add(entry);
+      }
+    }
+    return warnings;
+  }
+
+  public static Matcher<Iterable<Diagnostic<? extends JavaFileObject>>> onlyCastExceptions() {
+    return Matchers.everyItem(hasMessage(containsString("Type safety: Unchecked cast from")));
+  }
+
+  public static Matcher<Diagnostic<? extends JavaFileObject>> hasMessage( Matcher<String> messageMatcher ) {
+    return new FeatureMatcher<Diagnostic<? extends JavaFileObject>, String>(messageMatcher, "message", "message") {
+      @Override
+      protected String featureValueOf(Diagnostic<? extends JavaFileObject> actual) {
+        return actual.getMessage(Locale.getDefault());
```

Comment:
This is probably broken.  It could cause the project to test differently in different locales.